enhancing/enhancer.js

```diff
@@ -2,21 +2,29 @@ module.exports = {
   succeed,
   fail,
   repair,
-  get,
+  get
 };
 
 function succeed(item) {
-  return { ...item };
+  const enhancement = item.enhancement === 20 ? 20 : item.enhancement + 1;
+  return { ...item, enhancement };
 }
 
 function fail(item) {
-  return { ...item };
+  const enhancement =
+    item.enhancement > 16 ? item.enhancement - 1 : item.enhancement;
+  let durability =
+    item.enhancement < 15 ? item.durability - 5 : item.durability - 10;
+  durability = durability < 0 ? 0 : durability;
+  return { ...item, enhancement, durability };
 }
 
 function repair(item) {
-  return { ...item };
+  return { ...item, durability: 100 };
 }
 
 function get(item) {
-  return { ...item };
+  const name =
+    item.enhancement === 0 ? item.name : `[+${item.enhancement}] ${item.name}`;
```

Comment:
even more extra edge cases just because: what if a weapon has already been modified this way? you wouldn't want it to be named `[+7] [+7] Iron Dagger`